bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -822,10 +831,10 @@ public void run()
       };
     }
 
-    protected void release(boolean wait)
+    protected void release(boolean wait, boolean writer)
     {
       final int refCount = this.refCount.decrementAndGet();
-      if (refCount == 0 && storage != null) {
+      if (refCount == 0 && storage != null && canEvict(writer)) {
```

Comment:
Delegate entire condition check to canEvict(final int refCount, final boolean writer)? Move log.debug from the else condition on line 853 inside canEvict()?

Is writer true only for publisher?